### ISSUE TITLE
GMP build fails with gcc 4.9.1

### ISSUE THREAD

**Issue by alex-konovalov** [gap-system/gap#417]
After #399, building GMP on Ubuntu 14 machine with gcc 4.9.1 fails. This is a fragment of `bin/x86_64-pc-linux-gnu-gcc-default64/extern/gmp-6.0.0a/build_log`

```
libtool: link: g++ -Wall -g -O2 -m64 -m64 -g -O2 -m64 -o .libs/t-ops t-ops.o  -L../../.libs ../../tests/.libs/libtests.a ../../.libs/libgmpxx.so /home/hudson/workspace/GAP-dev-compilers/GAPCOPTS/64build/GAPGMP/gmp/GAPREADLINE/readline/label/gcc49/GAP-git-snapshot/extern/gmp-6.0.0a/.libs/libgmp.so ../../.libs/libgmp.so -lm -Wl,-rpath -Wl,/home/hudson/workspace/GAP-dev-compilers/GAPCOPTS/64build/GAPGMP/gmp/GAPREADLINE/readline/label/gcc49/GAP-git-snapshot/extern/gmp-6.0.0a/../../bin/x86_64-pc-linux-gnu-gcc-default64/extern/gmp-6.0.0a/lib
g++ -DHAVE_CONFIG_H -I. -I../.. -I../.. -I../../tests    -Wall -g -O2 -m64 -m64 -g -O2 -m64  -c -o t-ops2.o t-ops2.cc
```

This is the output of `gcc -v`:
```
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/4.9/lto-wrapper
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 4.9.1-3ubuntu2~14.04.1' --with-bugurl=file:///usr/share/doc/gcc-4.9/README.Bugs --enable-languages=c,c++,java,go,d,fortran,objc,obj-c++ --prefix=/usr --program-suffix=-4.9 --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --with-gxx-include-dir=/usr/include/c++/4.9 --libdir=/usr/lib --enable-nls --with-sysroot=/ --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --enable-gnu-unique-object --disable-vtable-verify --enable-plugin --with-system-zlib --disable-browser-plugin --enable-java-awt=gtk --enable-gtk-cairo --with-java-home=/usr/lib/jvm/java-1.5.0-gcj-4.9-amd64/jre --enable-java-home --with-jvm-root-dir=/usr/lib/jvm/java-1.5.0-gcj-4.9-amd64 --with-jvm-jar-dir=/usr/lib/jvm-exports/java-1.5.0-gcj-4.9-amd64 --with-arch-directory=amd64 --with-ecj-jar=/usr/share/java/eclipse-ecj.jar --enable-objc-gc --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 4.9.1 (Ubuntu 4.9.1-3ubuntu2~14.04.1) 
```

**Comment by ChrisJefferson** [comment]
Is there some more lines showing the error, or is that it?

**Comment by fingolfin** [comment]
Yeah, without the actual error message, this is quite useless. Can you perhaps just attach the full build log here?

**Comment by alex-konovalov** [comment]
Ah, probably copy-pasting error. Some lines were not included in the original report:
```
libtool: link: g++ -Wall -g -O2 -m64 -m64 -g -O2 -m64 -o .libs/t-ops t-ops.o  -L../../.libs ../../tests/.libs/libtests.a ../../.libs/libgmpxx.so /home/hudson/workspace/GAP-dev-compilers/GAPCOPTS/64build/GAPGMP/gmp/GAPREADLINE/readline/label/gcc49/GAP-git-snapshot/extern/gmp-6.0.0a/.libs/libgmp.so ../../.libs/libgmp.so -lm -Wl,-rpath -Wl,/home/hudson/workspace/GAP-dev-compilers/GAPCOPTS/64build/GAPGMP/gmp/GAPREADLINE/readline/label/gcc49/GAP-git-snapshot/extern/gmp-6.0.0a/../../bin/x86_64-pc-linux-gnu-gcc-default64/extern/gmp-6.0.0a/lib
g++ -DHAVE_CONFIG_H -I. -I../.. -I../.. -I../../tests    -Wall -g -O2 -m64 -m64 -g -O2 -m64  -c -o t-ops2.o t-ops2.cc
g++: internal compiler error: Killed (program cc1plus)
Please submit a full bug report,
with preprocessed source if appropriate.
See <file:///usr/share/doc/gcc-4.9/README.Bugs> for instructions.
```

**Comment by ChrisJefferson** [comment]
Argh. I don't mind telling people to go and just install g++, but if there are segfaults coming out on a regular basis, that's a bigger problem.

Could you try rebuilding? It could be the machine you are on is a bit wobbly (if we are lucky). We could also try 6.1.0.

**Comment by markuspf** [comment]
On Mon, Dec 21, 2015 at 02:21:26PM -0800, Alexander Konovalov wrote:
> Ah, probably copy-pasting error. Some lines were not included in the original
> report:
> 
> libtool: link: g++ -Wall -g -O2 -m64 -m64 -g -O2 -m64 -o .libs/t-ops t-ops.o  -L../../.libs ../../tests/.libs/libtests.a ../../.libs/libgmpxx.so /home/hudson/workspace/GAP-dev-compilers/GAPCOPTS/64build/GAPGMP/gmp/GAPREADLINE/readline/label/gcc49/GAP-git-snapshot/extern/gmp-6.0.0a/.libs/libgmp.so ../../.libs/libgmp.so -lm -Wl,-rpath -Wl,/home/hudson/workspace/GAP-dev-compilers/GAPCOPTS/64build/GAPGMP/gmp/GAPREADLINE/readline/label/gcc49/GAP-git-snapshot/extern/gmp-6.0.0a/../../bin/x86_64-pc-linux-gnu-gcc-default64/extern/gmp-6.0.0a/lib
> g++ -DHAVE_CONFIG_H -I. -I../.. -I../.. -I../../tests    -Wall -g -O2 -m64 -m64 -g -O2 -m64  -c -o t-ops2.o t-ops2.cc
> g++: internal compiler error: Killed (program cc1plus)
> Please submit a full bug report,
> with preprocessed source if appropriate.
> See <file:///usr/share/doc/gcc-4.9/README.Bugs> for instructions.

This sometimes happens for me if the compiler runs out of memory. FWIW, with

  ╰─❭ gcc --version
  gcc 5.3.1 [DragonFly] Release/2015-12-04

I don't get any crashes of this sort building GMP or GAP.
>

**Comment by alex-konovalov** [comment]
That was likely running out of memory - it sometimes happened before on these two hosts (OpenSUSE and Ubuntu 14) running in the cloud, but only it in HPC-GAP tests (and not at the build stage), and never in GAP 4 tests. 

The problem started precisely after #399 has been merged, because GAP 4 builds seem to require more memory too. 

I've restarted VMs and now all builds. Increased settings to use more memory on these instances from now on - let's see if they will be more stable.
